shared_model/interfaces/common_objects/peer.hpp

```diff
@@ -22,6 +24,12 @@ namespace shared_model {
        */
       virtual const interface::types::AddressType &address() const = 0;
 
+      /**
+       * @return Peer TLS certficate
+       */
+      virtual const boost::optional<interface::types::TLSCertificateType>
+          &tlsCertificate() const = 0;
```

Comment:
This should be added to
- `operator==`
- `toString()`, but IMO without printing the whole cert. Maybe just set/unset.